pkg/servicenameresolutiondetector/store/node_store_test.go

```diff
@@ -0,0 +1,359 @@
+			nodes: []*corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node2"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+			},
+			expectedResult: []metav1.Condition{
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
+			},
+			expectedError: false,
+		},
+		{
+			name: "List with not ready and stale nodes",
+			nodes: []*corev1.Node{
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node2"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionFalse, LastHeartbeatTime: metav1.Now()},
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{Name: "node3"},
+					Status: corev1.NodeStatus{
+						Conditions: []corev1.NodeCondition{
+							{Type: corev1.NodeReady, Status: corev1.ConditionTrue, LastHeartbeatTime: metav1.NewTime(time.Now().Add(-10 * time.Minute))},
+						},
+					},
+				},
+			},
+			expectedResult: []metav1.Condition{
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionTrue},
+				{Type: string(corev1.NodeReady), Status: metav1.ConditionUnknown},
+			},
+			expectedError: false,
+		},
+		{
+			name:           "Empty node list",
+			nodes:          []*corev1.Node{},
+			expectedResult: []metav1.Condition{},
+			expectedError:  false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			nodeInterfaces := make([]interface{}, len(tc.nodes))
+			for i, node := range tc.nodes {
+				nodeInterfaces[i] = node
+			}
+
+			err := nodeInformer.GetStore().Replace(nodeInterfaces, "")
+			assert.NoError(t, err)
+
+			result, err := store.ListAll()
+
+			if tc.expectedError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, len(tc.expectedResult), len(result))
+
+				expectedMap := make(map[string]int)
+				resultMap := make(map[string]int)
+
+				for _, cond := range tc.expectedResult {
+					key := fmt.Sprintf("%s:%s", cond.Type, cond.Status)
+					expectedMap[key]++
+				}
+
+				for _, cond := range result {
+					key := fmt.Sprintf("%s:%s", cond.Type, cond.Status)
+					resultMap[key]++
+				}
+
+				assert.Equal(t, expectedMap, resultMap)
+			}
+		})
+	}
+}
+
+func TestSetNodeStatusCondition(t *testing.T) {
```

Comment:
Hey @anujagrawal699 , do we have tested FindNodeStatusCondition function explicitly  ?

Reply:
@yashpandey06 But still the FindNodeStatusCondition function is being indirectly tested within the TestNodeStore function.